indico/modules/events/registration/fields/simple.py

```diff
@@ -304,18 +329,20 @@ def process_field_data(cls, data, old_data=None, old_versioned_data=None):
         unversioned_data['captions'] = captions
         return unversioned_data, versioned_data
 
-    @property
-    def view_data(self):
+    @classmethod
+    def unprocess_field_data(self, versioned_data, unversioned_data):
         data = {}
-        arrival_date_from = _to_date(self.form_item.data['arrival_date_from'])
-        arrival_date_to = _to_date(self.form_item.data['arrival_date_to'])
-        departure_date_from = _to_date(self.form_item.data['departure_date_from'])
-        departure_date_to = _to_date(self.form_item.data['departure_date_to'])
-        data['arrival_dates'] = [format_date(date) for date in iterdays(arrival_date_from, arrival_date_to)]
-        data['departure_dates'] = [format_date(date) for date in iterdays(departure_date_from, departure_date_to)]
-        items = deepcopy(self.form_item.versioned_data['choices'])
+        arrival_date_from = _to_date(unversioned_data['arrival_date_from'])
+        arrival_date_to = _to_date(unversioned_data['arrival_date_to'])
+        departure_date_from = _to_date(unversioned_data['departure_date_from'])
+        departure_date_to = _to_date(unversioned_data['departure_date_to'])
+        data['arrival_dates'] = [(dt.isoformat()[:10], format_date(dt)) for dt in
```

Comment:
Would converting to `.date().isoformat()` cause problems with timezones etc? If not I think that would be cleaner than truncating the datetime string